delta/sdk/src/test/scala/ch/epfl/bluebrain/nexus/delta/sdk/resources/ResourcesImplSpec.scala

```diff
@@ -410,24 +421,34 @@ class ResourcesImplSpec
     }
 
     "refreshing a resource" should {
+      val schema               = schemas.resources
+      val contextId            = nxv + "context"
+      val contextSource        = json""" { "@context": { "@base": "http://bbp.epfl.ch/base/" } }"""
+      val contextUpdatedSource =
+        json""" { "@context": { "@base": "http://example.com/base/", "prefix": "http://bbp.epfl.ch/prefix" } }"""
+      val idRefresh            = nxv + "refresh"
+
+      val refreshSource = json"""{
+                                    "@context": [
+                                      "$contextId",
+                                      { "@vocab": "https://bluebrain.github.io/nexus/vocabulary/" }
+                                    ],
+                                    "@type": "Person",
+                                    "name": "Alex"
+                                 }"""
 
       "succeed" in {
-        val expectedData =
-          ResourceGen.resource(myId6, projectRef, source.removeKeys(keywords.id), Revision(schema1.id, 1))
-        resources.refresh(myId6, projectRef, Some(schema1.id)).accepted shouldEqual
-          mkResource(expectedData).copy(rev = 2)
-      }
-
-      "succeed without specifying the schema" in {
-        val expectedData =
-          ResourceGen.resource(myId6, projectRef, source.removeKeys(keywords.id), Revision(schema1.id, 1))
-        resources.refresh("nxv:myid6", projectRef, None).accepted shouldEqual
-          ResourceGen.resourceFor(
-            expectedData,
-            types = types,
-            subject = subject,
-            rev = 3
-          )
+        val io = for {
+          _                      <- resources.create(contextId, projectRef, schema, contextSource, None)
+          createdResult          <- resources.create(idRefresh, projectRef, schema, refreshSource, None)
+          refreshedNoChange      <- resources.refresh(idRefresh, projectRef, None)
+          _                       = refreshedNoChange shouldEqual createdResult
+          updatedContext         <- resources.update(contextId, projectRef, None, 1, contextUpdatedSource, None)
+          _                       = updatedContext.rev shouldEqual 2
+          refreshedContextChange <- resources.refresh(idRefresh, projectRef, None)
+          _                       = refreshedContextChange shouldEqual createdResult.copy(rev = 2)
+        } yield ()
+        io.accepted
```

Comment:
This seems to be perhaps too many tests in one IMO

Reply:
The idea is to check that the refresh operation on the resource only creates a revision if one of the remote contexts has been updated.